lib/ansible/plugins/filter/regex_search.yml

```diff
@@ -32,6 +32,15 @@ EXAMPLES: |
   # drinkat => 'BAR'
   drinkat: "{{ 'foo\nBAR' | regex_search('^bar', multiline=True, ignorecase=True) }}"
 
+  # Extracts server and database id from a string
+  {{ 'server1/database42' | regex_search('server([0-9]+)/database([0-9]+)', '\\1', '\\2') }}
+  # => ['1', '42']
+
+  # Extracts dividend and divisor from a division
+  {{ '21/42' | regex_search('(?P<dividend>[0-9]+)/(?P<divisor>[0-9]+)', '\\g<dividend>', '\\g<divisor>') }}
+  # => ['21', '42']
```

Comment:
Since these examples are not specific to Ansible, there's no need to include specific examples here. Instead, it would be more appropriate to provide a link to the Python regex documentation. Otherwise we'll end up reproducing too many regex examples here.